e2e/setup/serializers/image/snapshot-matcher.ts

```diff
@@ -0,0 +1,94 @@
+import path from 'path';
+import fs from 'fs';
+import sharp from 'sharp';
+import pixelmatch from 'pixelmatch';
+
+import {DiffBuilder} from './diff-builder';
+
+export type SnapshotMatcherOptions = pixelmatch.PixelmatchOptions;
+
+export type SharpContext = {data: Buffer, info: sharp.OutputInfo};
+
+export class SnapshotMatcher {
+  protected static readonly DIFF_DIR = path.join(process.cwd(), '.diff');
+  protected static readonly SNAPSHOT_DIR = path.join(process.cwd(), 'tests', '__image_snapshots__');
+
+  protected static readonly defaultOptions: SnapshotMatcherOptions = {
+    diffMask: false,
+    alpha: 0,
+    diffColorAlt: [255, 255, 0],
+    includeAA: true,
+    threshold: 0.01
+  };
+
+  protected config: SnapshotMatcherOptions;
+  protected testName: string;
+  protected currentImg: sharp.Sharp;
+  protected shouldUpdateImg: boolean;
+
+  constructor(context: jest.MatcherContext, received: Buffer, options: SnapshotMatcherOptions = {}) {
+    this.testName = context.currentTestName!.replace(/([^a-z0-9]+)/gi, '-').toLowerCase();
+    this.config = Object.assign({}, SnapshotMatcher.defaultOptions, options);
+    this.currentImg = sharp(received).webp();
+    this.shouldUpdateImg = context.snapshotState._updateSnapshot === 'all';
+  }
+
+  public async match(): Promise<jest.CustomMatcherResult> {
+    this.updateDirectory(SnapshotMatcher.SNAPSHOT_DIR);
+    const prevImgPath = path.join(SnapshotMatcher.SNAPSHOT_DIR, `${this.testName}.webp`);
+    if (!fs.existsSync(prevImgPath) || this.shouldUpdateImg) {
+      await this.currentImg.toFile(prevImgPath);
+      return this.getMatcherResult(true, `New snapshot was created: ${prevImgPath}`);
+    }
+
+    const prevImg = await this.convertToRaw(prevImgPath);
+    const currImg = await this.convertToRaw(await this.currentImg.toBuffer());
+
+    let diffPath;
+    try {
+      diffPath = await this.compareImages(prevImg, currImg);
```

Comment:
It's an internal implementation, so it is a cosmetic, but  it is a little bit overmixed API for 
`compareImages` (in some cases we get error in some result)
I'd recommend to consider making it consistent - ideally safe-function :
 ```ts
 const diff = await this.compareImages(prevImg, currImg);
 
 if (!diff) return this.getMatcherResult(true, `Image is the same as the snapshot: ${prevImgPath}`);
 if (diff.reason = 'content') this.getMatcherResult(false, `Image mismatch found: ${diff.path}`);
 return this.getMatcherResult(false, `Error comparing snapshot to image ${prevImgPath}`);
 ```